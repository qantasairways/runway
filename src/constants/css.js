export const CSS_SELECTOR_ACTIVE = '&:active';
export const CSS_SELECTOR_HOVER = '&:hover';
export const CSS_SELECTOR_FOCUS = '&:focus';
export const CSS_SELECTOR_FIRSTCHILD = '&:first-child';
export const CSS_SELECTOR_LASTCHILD = '&:last-child';
export const CSS_PSEUDO_AFTER = '&::after';
export const CSS_PSEUDO_BEFORE = '&::before';
