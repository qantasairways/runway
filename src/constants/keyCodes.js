export const KEY_CODE_LEFT = 37;
export const KEY_CODE_RIGHT = 39;
export const KEY_CODE_UP = 38;
export const KEY_CODE_DOWN = 40;
export const KEY_CODE_SPACE = 32;
export const KEY_CODE_ENTER = 13;
export const KEY_CODE_TAB = 9;
