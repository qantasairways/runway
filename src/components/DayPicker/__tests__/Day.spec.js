import React from 'react';
import { shallow } from 'enzyme';
import noop from '../../../utils/noop';

import Day from '../components/Day';

import {
  getShouldSelectAsStartDate,
  getEndDateFromStartDate
} from '../helpers';
import { KEY_CODE_ENTER, KEY_CODE_SPACE } from '../../../constants/keyCodes';

jest.mock('../helpers', () => ({
  getShouldSelectAsStartDate: jest.fn(),
  getEndDateFromStartDate: jest.fn(),
  DAY_CELL_HEIGHT_DESKTOP: 90,
  DAY_CELL_HEIGHT_MOBILE: 94
}));

const DummyIcon = () => <div>Icon</div>;

describe('Day', () => {
  let component;

  const date = new Date(2019, 3, 20, 0, 0, 0);
  const defaultProps = {
    date,
    timestamp: date.getTime(),
    month: 'January',
    year: 2018,
    onDayClick: noop,
    onDayNavigate: noop
  };

  const props = {
    isSelectingStartDate: false,
    startSelectedLabel: 'MEL',
    endSelectedLabel: 'SYD',
    startLabel: 'Depart',
    endLabel: 'Return',
    startAriaLabel: 'Selected for departure',
    endAriaLabel: 'Selected for return',
    Icon: DummyIcon,
    isDesktopDevice: false,
    startDate: new Date(2019, 3, 21, 0, 0, 0),
    endDate: new Date(2019, 3, 23, 0, 0, 0),
    isStart: false,
    isEnd: false,
    isInRange: false,
    isDisabled: false,
    isOutside: false,
    isToday: false,
    isFirstDayOfMonth: false,
    isDateRange: true,
    isLoadingPrice: false,
    currencyCode: 'AUD',
    currencySymbol: '$',
    price: null
  };

  it('renders correctly with defaults', () => {
    component = shallow(<Day {...defaultProps} />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly with props provided', () => {
    component = shallow(<Day {...defaultProps} {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly when isStart is true', () => {
    component = shallow(<Day {...defaultProps} {...props} isStart />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly when isEnd is true', () => {
    component = shallow(<Day {...defaultProps} {...props} isEnd />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly when isInRange is true', () => {
    component = shallow(<Day {...defaultProps} {...props} isInRange />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly when isDisabled is true', () => {
    component = shallow(<Day {...defaultProps} {...props} isDisabled />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly when isOutside is true', () => {
    component = shallow(<Day {...defaultProps} {...props} isOutside />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly when isToday is true', () => {
    component = shallow(<Day {...defaultProps} {...props} isToday />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly when isFirstDayOfMonth is true', () => {
    component = shallow(<Day {...defaultProps} {...props} isFirstDayOfMonth />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly when isLoadingPrice is true', () => {
    component = shallow(<Day {...defaultProps} {...props} isLoadingPrice />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly when price is provided', () => {
    component = shallow(
      <Day
        {...defaultProps}
        {...props}
        price={{
          value: 123,
          taxValue: 2,
          isLowestPrice: false
        }}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('renders correctly when isDesktopDevice is true and this.state.hover is true', () => {
    component = shallow(<Day {...defaultProps} {...props} isDesktopDevice />);
    component.instance().setState({ hover: true });

    expect(component).toMatchSnapshot();
  });

  describe('renderHoverLabel()', () => {
    it('returns null if isDisabled is true', () => {
      component = shallow(<Day {...defaultProps} {...props} isDisabled />);

      expect(component.instance().renderHoverLabel()).toBe(null);
    });

    it('returns null if isSelectingStartDate is true and isStart is true', () => {
      component = shallow(
        <Day {...defaultProps} {...props} isSelectingStartDate isStart />
      );

      expect(component.instance().renderHoverLabel()).toBe(null);
    });

    it('returns null if isSelectingStartDate is false and isEnd is true', () => {
      component = shallow(
        <Day {...defaultProps} {...props} isSelectingStartDate={false} isEnd />
      );

      expect(component.instance().renderHoverLabel()).toBe(null);
    });

    describe('when getShouldSelectAsStartDate returns true', () => {
      beforeEach(() => {
        getShouldSelectAsStartDate.mockImplementation(() => true);
      });

      it('returns the correct markup', () => {
        component = shallow(<Day {...defaultProps} {...props} />);

        expect(component.instance().renderHoverLabel()).toMatchSnapshot();
      });
    });

    describe('when getShouldSelectAsStartDate returns false', () => {
      beforeEach(() => {
        getShouldSelectAsStartDate.mockImplementation(() => false);
      });

      it('returns the correct markup when isEnd is false', () => {
        component = shallow(<Day {...defaultProps} {...props} isEnd={false} />);

        expect(component.instance().renderHoverLabel()).toMatchSnapshot();
      });

      it('returns null when isEnd is true', () => {
        component = shallow(<Day {...defaultProps} {...props} isEnd />);

        expect(component.instance().renderHoverLabel()).toBe(null);
      });
    });
  });

  describe('handleKeyDown()', () => {
    const onDayClickMock = jest.fn();
    const onDayNavigateMock = jest.fn();
    const preventDefaultMock = jest.fn();
    const stopPropagationMock = jest.fn();

    afterEach(() => {
      onDayClickMock.mockReset();
      onDayNavigateMock.mockReset();
      preventDefaultMock.mockReset();
      stopPropagationMock.mockReset();
    });

    it('calls handleDayClick when key is the Enter key', () => {
      component = shallow(
        <Day {...defaultProps} {...props} onDayClick={onDayClickMock} />
      );
      component.instance().handleKeyDown({
        keyCode: KEY_CODE_ENTER,
        preventDefault: preventDefaultMock,
        stopPropagation: stopPropagationMock
      });
      expect(onDayClickMock.mock.calls.length).toBe(1);
      expect(preventDefaultMock.mock.calls.length).toBe(1);
      expect(stopPropagationMock.mock.calls.length).toBe(1);
    });

    it('calls handleDayClick when key is the Space key', () => {
      component = shallow(
        <Day {...defaultProps} {...props} onDayClick={onDayClickMock} />
      );
      component.instance().handleKeyDown({
        keyCode: KEY_CODE_SPACE,
        preventDefault: preventDefaultMock,
        stopPropagation: stopPropagationMock
      });
      expect(onDayClickMock.mock.calls.length).toBe(1);
      expect(preventDefaultMock.mock.calls.length).toBe(1);
      expect(stopPropagationMock.mock.calls.length).toBe(1);
    });

    it('calls this.props.onDayNavigate with correct args', () => {
      component = shallow(
        <Day {...defaultProps} {...props} onDayNavigate={onDayNavigateMock} />
      );
      component.instance().handleKeyDown({ keyCode: 555 });
      expect(onDayNavigateMock.mock.calls.length).toBe(1);
      expect(onDayNavigateMock.mock.calls[0][0]).toBe(date.getTime());
      expect(onDayNavigateMock.mock.calls[0][1]).toBe(555);
    });
  });

  describe('handleMouseOver()', () => {
    it('sets this.state.hover to true', () => {
      component = shallow(<Day {...defaultProps} {...props} />);
      component.instance().handleMouseOver();
      expect(component.instance().state.hover).toBe(true);
    });
  });

  describe('handleMouseLeave()', () => {
    it('sets this.state.hover to false', () => {
      component = shallow(<Day {...defaultProps} {...props} />);
      component.instance().setState({ hover: true });
      component.instance().handleMouseLeave();
      expect(component.instance().state.hover).toBe(false);
    });
  });

  describe('handleDayClick()', () => {
    const onDayClickMock = jest.fn();
    const calculatedEndDate = new Date(2019, 10, 11, 0, 0, 0, 0);

    afterEach(() => {
      onDayClickMock.mockReset();
      getShouldSelectAsStartDate.mockReset();
      getEndDateFromStartDate.mockReset();
    });

    it('sets this.state.hover to false', () => {
      component = shallow(<Day {...defaultProps} {...props} />);
      component.instance().setState({ hover: true });
      component.instance().handleDayClick();
      expect(component.instance().state.hover).toBe(false);
    });

    it('does not call this.props.onDayClick when the day is disabled', () => {
      component = shallow(
        <Day
          {...defaultProps}
          {...props}
          isDisabled
          onDayClick={onDayClickMock}
        />
      );
      component.instance().setState({ hover: true });
      component.instance().handleDayClick();
      expect(onDayClickMock.mock.calls.length).toBe(0);
      expect(component.instance().state.hover).toBe(false);
    });

    describe('when getShouldSelectAsStartDate returns true', () => {
      beforeEach(() => {
        getShouldSelectAsStartDate.mockImplementation(() => true);
        getEndDateFromStartDate.mockImplementation(() => calculatedEndDate);
      });

      it('calls this.props.onDayClick with the correct args', () => {
        component = shallow(
          <Day {...defaultProps} {...props} onDayClick={onDayClickMock} />
        );
        component.instance().handleDayClick();
        expect(onDayClickMock.mock.calls.length).toBe(1);
        expect(onDayClickMock.mock.calls[0][0]).toBe(date);
        expect(onDayClickMock.mock.calls[0][1]).toBe(calculatedEndDate);
        expect(onDayClickMock.mock.calls[0][2]).toBe(false);
      });

      it('calls this.props.onDayClick with the third argument as true when this.props.isDateRange is false', () => {
        component = shallow(
          <Day
            {...defaultProps}
            {...props}
            onDayClick={onDayClickMock}
            isDateRange={false}
          />
        );
        component.instance().handleDayClick();
        expect(onDayClickMock.mock.calls.length).toBe(1);
        expect(onDayClickMock.mock.calls[0][0]).toBe(date);
        expect(onDayClickMock.mock.calls[0][1]).toBe(calculatedEndDate);
        expect(onDayClickMock.mock.calls[0][2]).toBe(true);
      });
    });

    describe('when getShouldSelectAsStartDate returns false', () => {
      beforeEach(() => {
        getShouldSelectAsStartDate.mockImplementation(() => false);
      });

      it('calls this.props.onDayClick with the correct args', () => {
        component = shallow(
          <Day {...defaultProps} {...props} onDayClick={onDayClickMock} />
        );
        component.instance().handleDayClick();
        expect(onDayClickMock.mock.calls.length).toBe(1);
        expect(onDayClickMock.mock.calls[0][0]).toBe(props.startDate);
        expect(onDayClickMock.mock.calls[0][1]).toBe(date);
        expect(onDayClickMock.mock.calls[0][2]).toBe(true);
      });
    });
  });
});
