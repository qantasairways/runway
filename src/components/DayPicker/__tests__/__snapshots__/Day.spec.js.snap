// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day renderHoverLabel() when getShouldSelectAsStartDate returns false returns the correct markup when isEnd is false 1`] = `
<DayLabel
  Icon={[Function]}
  bottomAligned={false}
  isSelected={false}
  label="Return"
  leftAligned={false}
/>
`;

exports[`Day renderHoverLabel() when getShouldSelectAsStartDate returns true returns the correct markup 1`] = `
<DayLabel
  Icon={[Function]}
  bottomAligned={false}
  isSelected={false}
  label="Depart"
  leftAligned={true}
/>
`;

exports[`Day renders correctly when isDesktopDevice is true and this.state.hover is true 1`] = `
.emotion-1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  padding-top: 36px;
  box-sizing: border-box;
  height: 94px;
  color: #323232;
  background: #fff;
  box-shadow: 0 0 0 1px #eaeaea;
  outline: none;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    padding-top: 33px;
    height: 90px;
  }
}

.emotion-1:hover,
.emotion-1:focus {
  z-index: 2;
  box-shadow: -2px 3px 4px 0 rgba(0,0,0,0.08),2px -3px 4px 0 rgba(0,0,0,0.04);
}

.emotion-1:hover div[class$="runway-calendar__hover-label"] {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  height: 1.125rem;
  padding: 2px 4px;
  font-size: 1.125rem;
  line-height: 1.1;
  border-radius: 4px;
  box-sizing: content-box;
  background-color: initial;
  color: inherit;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    font-size: 1.375rem;
    line-height: 0.95;
  }
}

<div
  aria-label="20 January 2018. Selected for return"
  className="emotion-1 d1555682400000"
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  role="button"
  tabIndex={-1}
>
  <div
    className="emotion-0"
  >
    20
  </div>
  <DayLabel
    Icon={[Function]}
    bottomAligned={false}
    isSelected={false}
    label="Return"
    leftAligned={false}
  />
</div>
`;

exports[`Day renders correctly when isDisabled is true 1`] = `
.emotion-0 {
  height: 1.125rem;
  padding: 2px 4px;
  font-size: 1.125rem;
  line-height: 1.1;
  border-radius: 4px;
  box-sizing: content-box;
  background-color: initial;
  color: inherit;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    font-size: 1.375rem;
    line-height: 0.95;
  }
}

.emotion-1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  padding-top: 36px;
  box-sizing: border-box;
  height: 94px;
  color: #323232;
  background: #fff;
  box-shadow: 0 0 0 1px #eaeaea;
  outline: none;
  cursor: initial;
  color: #dadada;
  background: none;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    padding-top: 33px;
    height: 90px;
  }
}

<div
  aria-label="20 January 2018. Selected for return"
  className="emotion-1 "
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  role="button"
  tabIndex={-1}
>
  <div
    className="emotion-0"
  >
    20
  </div>
</div>
`;

exports[`Day renders correctly when isEnd is true 1`] = `
.emotion-0 {
  height: 1.125rem;
  padding: 2px 4px;
  font-size: 1.125rem;
  line-height: 1.1;
  border-radius: 4px;
  box-sizing: content-box;
  background-color: initial;
  color: inherit;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    font-size: 1.375rem;
    line-height: 0.95;
  }
}

.emotion-1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  padding-top: 36px;
  box-sizing: border-box;
  height: 94px;
  color: #323232;
  background: #fff;
  box-shadow: 0 0 0 1px #eaeaea;
  outline: none;
  padding-top: 34px;
  z-index: 1;
  border: 2px solid #bff4f2;
  box-shadow: 0 0 0 1px #bff4f2;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    padding-top: 33px;
    height: 90px;
  }
}

.emotion-1:hover,
.emotion-1:focus {
  z-index: 2;
  box-shadow: -2px 3px 4px 0 rgba(0,0,0,0.08),2px -3px 4px 0 rgba(0,0,0,0.04);
}

.emotion-1:hover div[class$="runway-calendar__hover-label"] {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    padding-top: 31px;
  }
}

<div
  aria-label="Return SYD. 20 January 2018. Selected for return"
  className="emotion-1 d1555682400000"
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  role="button"
  tabIndex={-1}
>
  <div
    className="emotion-0"
  >
    20
  </div>
  <DayLabel
    Icon={[Function]}
    bottomAligned={false}
    isSelected={true}
    label="SYD"
    leftAligned={false}
  />
</div>
`;

exports[`Day renders correctly when isFirstDayOfMonth is true 1`] = `
.emotion-1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  padding-top: 36px;
  box-sizing: border-box;
  height: 94px;
  color: #323232;
  background: #fff;
  box-shadow: 0 0 0 1px #eaeaea;
  outline: none;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    padding-top: 33px;
    height: 90px;
  }
}

.emotion-1:hover,
.emotion-1:focus {
  z-index: 2;
  box-shadow: -2px 3px 4px 0 rgba(0,0,0,0.08),2px -3px 4px 0 rgba(0,0,0,0.04);
}

.emotion-1:hover div[class$="runway-calendar__hover-label"] {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  height: 1.125rem;
  padding: 2px 4px;
  font-size: 1.125rem;
  line-height: 1.1;
  border-radius: 4px;
  box-sizing: content-box;
  background-color: initial;
  color: inherit;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    font-size: 1.375rem;
    line-height: 0.95;
  }
}

<div
  aria-label="20 January 2018. Selected for return"
  className="emotion-1 d1555682400000"
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="emotion-0"
  >
    20
  </div>
</div>
`;

exports[`Day renders correctly when isInRange is true 1`] = `
.emotion-0 {
  height: 1.125rem;
  padding: 2px 4px;
  font-size: 1.125rem;
  line-height: 1.1;
  border-radius: 4px;
  box-sizing: content-box;
  background-color: initial;
  color: inherit;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    font-size: 1.375rem;
    line-height: 0.95;
  }
}

.emotion-1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  padding-top: 36px;
  box-sizing: border-box;
  height: 94px;
  color: #323232;
  background: #fff;
  box-shadow: 0 0 0 1px #eaeaea;
  outline: none;
  background-color: #e8f7f7;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    padding-top: 33px;
    height: 90px;
  }
}

.emotion-1:hover,
.emotion-1:focus {
  z-index: 2;
  box-shadow: -2px 3px 4px 0 rgba(0,0,0,0.08),2px -3px 4px 0 rgba(0,0,0,0.04);
}

.emotion-1:hover div[class$="runway-calendar__hover-label"] {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  aria-label="20 January 2018. Selected for return"
  className="emotion-1 d1555682400000"
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  role="button"
  tabIndex={-1}
>
  <div
    className="emotion-0"
  >
    20
  </div>
</div>
`;

exports[`Day renders correctly when isLoadingPrice is true 1`] = `
.emotion-1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  padding-top: 36px;
  box-sizing: border-box;
  height: 94px;
  color: #323232;
  background: #fff;
  box-shadow: 0 0 0 1px #eaeaea;
  outline: none;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    padding-top: 33px;
    height: 90px;
  }
}

.emotion-1:hover,
.emotion-1:focus {
  z-index: 2;
  box-shadow: -2px 3px 4px 0 rgba(0,0,0,0.08),2px -3px 4px 0 rgba(0,0,0,0.04);
}

.emotion-1:hover div[class$="runway-calendar__hover-label"] {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  height: 1.125rem;
  padding: 2px 4px;
  font-size: 1.125rem;
  line-height: 1.1;
  border-radius: 4px;
  box-sizing: content-box;
  background-color: initial;
  color: inherit;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    font-size: 1.375rem;
    line-height: 0.95;
  }
}

<div
  aria-label="20 January 2018. Selected for return"
  className="emotion-1 d1555682400000"
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  role="button"
  tabIndex={-1}
>
  <div
    className="emotion-0"
  >
    20
  </div>
  <Price
    currencySymbol="$"
    isDesktopDevice={false}
    isLoadingPrice={true}
    isLowestPrice={false}
    taxValue={null}
    value={null}
  />
</div>
`;

exports[`Day renders correctly when isOutside is true 1`] = `
<div
  key="1555682400000"
/>
`;

exports[`Day renders correctly when isStart is true 1`] = `
.emotion-0 {
  height: 1.125rem;
  padding: 2px 4px;
  font-size: 1.125rem;
  line-height: 1.1;
  border-radius: 4px;
  box-sizing: content-box;
  background-color: initial;
  color: inherit;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    font-size: 1.375rem;
    line-height: 0.95;
  }
}

.emotion-1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  padding-top: 36px;
  box-sizing: border-box;
  height: 94px;
  color: #323232;
  background: #fff;
  box-shadow: 0 0 0 1px #eaeaea;
  outline: none;
  padding-top: 34px;
  z-index: 1;
  border: 2px solid #bff4f2;
  box-shadow: 0 0 0 1px #bff4f2;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    padding-top: 33px;
    height: 90px;
  }
}

.emotion-1:hover,
.emotion-1:focus {
  z-index: 2;
  box-shadow: -2px 3px 4px 0 rgba(0,0,0,0.08),2px -3px 4px 0 rgba(0,0,0,0.04);
}

.emotion-1:hover div[class$="runway-calendar__hover-label"] {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    padding-top: 31px;
  }
}

<div
  aria-label="Depart MEL. 20 January 2018. Selected for return"
  className="emotion-1 d1555682400000"
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="emotion-0"
  >
    20
  </div>
  <DayLabel
    Icon={[Function]}
    bottomAligned={false}
    isSelected={true}
    label="MEL"
    leftAligned={true}
  />
</div>
`;

exports[`Day renders correctly when isToday is true 1`] = `
.emotion-1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  padding-top: 36px;
  box-sizing: border-box;
  height: 94px;
  color: #323232;
  background: #fff;
  box-shadow: 0 0 0 1px #eaeaea;
  outline: none;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    padding-top: 33px;
    height: 90px;
  }
}

.emotion-1:hover,
.emotion-1:focus {
  z-index: 2;
  box-shadow: -2px 3px 4px 0 rgba(0,0,0,0.08),2px -3px 4px 0 rgba(0,0,0,0.04);
}

.emotion-1:hover div[class$="runway-calendar__hover-label"] {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  height: 1.125rem;
  padding: 2px 4px;
  font-size: 1.125rem;
  line-height: 1.1;
  border-radius: 4px;
  box-sizing: content-box;
  background-color: #bff4f2;
  color: inherit;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    font-size: 1.375rem;
    line-height: 0.95;
  }
}

<div
  aria-label="20 January 2018. Selected for return"
  className="emotion-1 d1555682400000"
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  role="button"
  tabIndex={-1}
>
  <div
    className="emotion-0"
  >
    20
  </div>
</div>
`;

exports[`Day renders correctly when price is provided 1`] = `
.emotion-1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  padding-top: 36px;
  box-sizing: border-box;
  height: 94px;
  color: #323232;
  background: #fff;
  box-shadow: 0 0 0 1px #eaeaea;
  outline: none;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    padding-top: 33px;
    height: 90px;
  }
}

.emotion-1:hover,
.emotion-1:focus {
  z-index: 2;
  box-shadow: -2px 3px 4px 0 rgba(0,0,0,0.08),2px -3px 4px 0 rgba(0,0,0,0.04);
}

.emotion-1:hover div[class$="runway-calendar__hover-label"] {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  height: 1.125rem;
  padding: 2px 4px;
  font-size: 1.125rem;
  line-height: 1.1;
  border-radius: 4px;
  box-sizing: content-box;
  background-color: initial;
  color: inherit;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    font-size: 1.375rem;
    line-height: 0.95;
  }
}

<div
  aria-label="20 January 2018. Selected for return"
  className="emotion-1 d1555682400000"
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  role="button"
  tabIndex={-1}
>
  <div
    className="emotion-0"
  >
    20
  </div>
  <Price
    currencySymbol="$"
    isDesktopDevice={false}
    isLoadingPrice={false}
    isLowestPrice={false}
    taxValue={2}
    value={123}
  />
</div>
`;

exports[`Day renders correctly with defaults 1`] = `
.emotion-1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  padding-top: 36px;
  box-sizing: border-box;
  height: 94px;
  color: #323232;
  background: #fff;
  box-shadow: 0 0 0 1px #eaeaea;
  outline: none;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    padding-top: 33px;
    height: 90px;
  }
}

.emotion-1:hover,
.emotion-1:focus {
  z-index: 2;
  box-shadow: -2px 3px 4px 0 rgba(0,0,0,0.08),2px -3px 4px 0 rgba(0,0,0,0.04);
}

.emotion-1:hover div[class$="runway-calendar__hover-label"] {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  height: 1.125rem;
  padding: 2px 4px;
  font-size: 1.125rem;
  line-height: 1.1;
  border-radius: 4px;
  box-sizing: content-box;
  background-color: initial;
  color: inherit;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    font-size: 1.375rem;
    line-height: 0.95;
  }
}

<div
  aria-label="20 January 2018. "
  className="emotion-1 d1555682400000"
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  role="button"
  tabIndex={-1}
>
  <div
    className="emotion-0"
  >
    20
  </div>
</div>
`;

exports[`Day renders correctly with props provided 1`] = `
.emotion-1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  padding-top: 36px;
  box-sizing: border-box;
  height: 94px;
  color: #323232;
  background: #fff;
  box-shadow: 0 0 0 1px #eaeaea;
  outline: none;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    padding-top: 33px;
    height: 90px;
  }
}

.emotion-1:hover,
.emotion-1:focus {
  z-index: 2;
  box-shadow: -2px 3px 4px 0 rgba(0,0,0,0.08),2px -3px 4px 0 rgba(0,0,0,0.04);
}

.emotion-1:hover div[class$="runway-calendar__hover-label"] {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  height: 1.125rem;
  padding: 2px 4px;
  font-size: 1.125rem;
  line-height: 1.1;
  border-radius: 4px;
  box-sizing: content-box;
  background-color: initial;
  color: inherit;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    font-size: 1.375rem;
    line-height: 0.95;
  }
}

<div
  aria-label="20 January 2018. Selected for return"
  className="emotion-1 d1555682400000"
  onBlur={[Function]}
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  role="button"
  tabIndex={-1}
>
  <div
    className="emotion-0"
  >
    20
  </div>
</div>
`;
